lib/src/main/java/com/futurewei/alcor/common/db/DistributedLockFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.common.db;
+
+import com.futurewei.alcor.common.db.ignite.IgniteDistributedLock;
+import com.futurewei.alcor.common.db.redis.RedisDistributedLock;
+import org.apache.ignite.client.IgniteClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.stereotype.Component;
+
+@ComponentScan
+@Component
+public class DistributedLockFactory {
+    @Autowired(required = false)
+    private IgniteClient igniteClient;
+
+    @Value("${lock.try.interval}")

```

Comment:
Quick question: I see these two configurations are added to the pom file of AlcorLib which means that it will applied to multiple microservices. Do you expect the need of customizing it differently?

Reply:
No, it is no need to set different values at those two configurations. I am just not familiar with the usage of configuration file...

Reply:
oh okay. Got it :-)

Reply:
Done. Changed to use default values instead of modifying all configuration files